util/tracing/detect/detect.go

```diff
@@ -52,17 +55,45 @@ func Register(name string, exp ExporterDetector, priority int) {
 	}
 }
 
+type SpanExporterDetector func() (sdktrace.SpanExporter, error)
+
+func (fn SpanExporterDetector) DetectSpanExporter() (sdktrace.SpanExporter, error) {
+	return fn()
+}
+
+func (fn SpanExporterDetector) DetectMetricExporter() (sdkmetric.Exporter, error) {
+	return nil, nil
+}
+
 func detectExporter() (texp sdktrace.SpanExporter, mexp sdkmetric.Exporter, err error) {
-	if n := os.Getenv("OTEL_TRACES_EXPORTER"); n != "" {
+	texp, err = detectExporterHelper("OTEL_TRACES_EXPORTER", func(d ExporterDetector) (sdktrace.SpanExporter, bool, error) {
```

Comment:
Maybe this should be `detectExporter` and current function should be `detectExporters`

Reply:
Done.